python/cudf/cudf/tests/pytest.ini

```diff
@@ -8,6 +8,8 @@ filterwarnings =
     error
     ignore:::.*xdist.*
     ignore:::.*pytest.*
+    # some third-party dependencies (e.g. 'boto3') still using datetime.datetime.utcnow()
+    ignore:.*datetime.*utcnow.*scheduled for removal.*:DeprecationWarning
```

Comment:
In the `cudf` and `dask_cudf` tests on Python 3.12, there are a few dozen cases where `DeprecationWarning` like the following are treated as errors

> DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).

([build link](https://github.com/rapidsai/cudf/actions/runs/10724599183/job/29742782194#step:7:4834))

Looking through the logs, those all appear to be coming from code in `boto3` / `botocore` that we don't control. There are no direct uses of `datetime.datetime.utcnow()` in this repo.

```shell
git grep utcnow
```

So this is proposing not failing tests when those warnings are raised.

Context on the deprecation of `datetime.datetime.utcnow()`:

* https://blog.miguelgrinberg.com/post/it-s-time-for-a-change-datetime-utcnow-is-now-deprecated
* https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow

Reply:
Is it possible to scope this more narrowly? I haven't looked at the error logs to see how widespread this problem is, but I wonder if we can use something like a local context manager to wrap the places where this is occurring.

Let's not treat this as a blocker for this PR -- but we might want to make it a follow-up task.

Reply:
I'm not sure. It's coming from this codepath in `botocore`

```text
self = <botocore.auth.S3SigV4Auth object at 0x7f6feb62f380>
request = <botocore.awsrequest.AWSRequest object at 0x7f700a3b3800>

    def add_auth(self, request):
        if self.credentials is None:
            raise NoCredentialsError()
>       datetime_now = datetime.datetime.utcnow()
E       DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
```

https://github.com/rapidsai/cudf/actions/runs/10724599183/job/29742782194#step:7:4279

If it's possible to ignore warnings from specific dependencies, that'd be useful. I couldn't figure out how to do that in a few minutes of googling and docs reading.

Reply:
> If it's possible to ignore warnings from specific dependencies, that'd be useful. I couldn't figure out how to do that in a few minutes of googling and docs reading.

In a warning filter, after specifying the warning category you can also specify `:module:line` https://docs.python.org/3/library/warnings.html#describing-warning-filters

So in a follow up PR, you can add `:botocore` at the end to just ignore this warning for that module

Reply:
Thanks @mroeschke , I finally went back and did this. You suggestion worked perfectly 😁 

https://github.com/rapidsai/cudf/pull/16896